src/mongocrypt-ctx-encrypt.c

```diff
@@ -938,6 +938,75 @@ generate_delete_tokens (_mongocrypt_crypto_t *crypto,
    return out;
 }
 
+static bool
+_check_for_payload_requiring_encryptionInformation (void *ctx,
+                                                    _mongocrypt_buffer_t *in,
+                                                    mongocrypt_status_t *status)
+{
+   bool *out = (bool *) ctx;
+
+   if (in->len < 1) {
+      CLIENT_ERR ("unexpected empty FLE payload");
+      return false;
+   }
+
+   if (in->data[0] == MC_SUBTYPE_FLE2InsertUpdatePayload) {
+      *out = true;
+      return true;
+   }
+
+   if (in->data[0] == MC_SUBTYPE_FLE2FindEqualityPayload) {
+      *out = true;
+      return true;
+   }
+
+   return true;
+}
+
+static bool
+must_omit_encryptionInformation (const char *command_name,
+                                 const bson_t *command,
+                                 bool *must_omit /* out */,
+                                 mongocrypt_status_t *status)
+{
+   const char *eligible_commands[] = {
+      "find", "aggregate", "distinct", "count", "insert"};
```

Comment:
@markbenvenuto are these conditions for omitting `encryptionInformation` correct?

1. The command is `find`, `aggregate`, `distinct`, `count`, or `insert.
2. The command has no `FLE2FindEncryptedPayload` or `FLE2InsertUpdatePayload`.

Reply:
Confirmed over slack.